test/crypto/CashTransactionsPaymentSearchEncrypterSpec.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package crypto
+
+import models.*
+import models.response.{
+  CashAccountTransactionSearchResponseDetail, DeclarationSearch, DeclarationWrapper,
+  EoriData, EoriDataContainer, PaymentType, PaymentsWithdrawalsAndTransfer, PaymentsWithdrawalsAndTransferContainer,
+  TaxGroupSearch, TaxGroupWrapper, TaxTypeWithSecurity, TaxTypeWithSecurityContainer
+}
+import utils.SpecBase
+
+class CashTransactionsPaymentSearchEncrypterSpec extends SpecBase {
+
+  private val cipher = new AesGCMCrypto
+  private val encrypter = new CashAccountPaymentSearchEncrypter(cipher)
+  private val secretKey = "VqmXp7yigDFxbCUdDdNZVIvbW6RgPNJsliv6swQNCL8="
+
+
+  "encrypt / decrypt cashTransactions" must {
+
+    "encrypt and decrypt declaration search results correctly" in new Setup {
```

Comment:
Better to rename it to as below as we are dealing with response details

"encrypt and decrypt declaration search response details correctly"